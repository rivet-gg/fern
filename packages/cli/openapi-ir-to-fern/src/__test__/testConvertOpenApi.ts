import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { parse } from "@fern-api/openapi-parser";
import { createMockTaskContext } from "@fern-api/task-context";
import path from "path";
import { convert } from "../convert";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testConvertOpenAPI(fixtureName: string, filename: string, asyncApiFilename?: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("simple", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: false,
                detectGlobalHeaders: true
            });
            expect(fernDefinition).toMatchSnapshot();
        });

        it("docs", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: true,
                detectGlobalHeaders: false
            });
            expect(fernDefinition).toMatchSnapshot();
        });
    });
}
