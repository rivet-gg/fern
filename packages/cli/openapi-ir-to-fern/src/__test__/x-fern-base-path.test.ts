import { testConvertOpenAPI } from "./testConvertOpenApi";

describe("x-fern-base-path", () => {
    testConvertOpenAPI("x-fern-base-path", "openapi.yml");
    testConvertOpenAPI("x-fern-base-path", "swagger.yml");
});
