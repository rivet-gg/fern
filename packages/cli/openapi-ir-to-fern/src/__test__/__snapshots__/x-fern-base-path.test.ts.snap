// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-fern-base-path x-fern-base-path docs 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {},
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Acme API",
    "environments": {
      "Default": "https://api.acme.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`x-fern-base-path x-fern-base-path docs 2`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "exampleEndpoint": {
          "auth": false,
          "display-name": "Example endpoint",
          "docs": undefined,
          "examples": [
            {},
          ],
          "method": "GET",
          "pagination": undefined,
          "path": "/v1/example",
        },
      },
    },
  },
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Acme API",
    "environments": {
      "Default": "https://api.acme.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`x-fern-base-path x-fern-base-path simple 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {},
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Acme API",
    "environments": {
      "Default": "https://api.acme.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`x-fern-base-path x-fern-base-path simple 2`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "exampleEndpoint": {
          "auth": false,
          "display-name": "Example endpoint",
          "docs": undefined,
          "examples": [
            {},
          ],
          "method": "GET",
          "pagination": undefined,
          "path": "/v1/example",
        },
      },
    },
  },
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Acme API",
    "environments": {
      "Default": "https://api.acme.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;
